File: simple/java/extract-method_isolate-switch_after.java
class Order {
  // ...

  public double calculateTotal() {
    double total = 0;
    for (Product product : getProducts()) {
      total = product.quantity * product.price;
    }
    total = applyRegionalDiscounts(total);
    return total;
  }

  public double applyRegionalDiscounts(total) {
    double result = total;
    switch (user.getCountry()) {
      case "US": result *= 0.85; break;
      case "RU": result *= 0.75; break;
      case "CN": result *= 0.9; break;
      // ...
    }
    return result;
  }

File: simple/php/extract-method_isolate-switch_after.php
<?php
class Order {
  // ...

  public function calculateTotal() {
    $total = 0;
    foreach ($this->getProducts() as $product) {
      $total = $product->quantity * $product->price;
    }
    $total = $this->applyRegionalDiscounts($total);
    return $total;
  }

  public function applyRegionalDiscounts($total) {
    $result = $total;
    switch ($this->user->getCountry()) {
      case "US": $result *= 0.85; break;
      case "RU": $result *= 0.75; break;
      case "CN": $result *= 0.9; break;
      // ...
    }
    return $result;
  }